File: src/main/java/com/vc/easy/L415.scala
package com.vc.easy

object L415 {
  def addStrings(num1: String, num2: String): String = {
    import scala.collection.mutable
    var i = num1.length - 1
    var j = num2.length - 1
    var carry = 0
    val str = new mutable.StringBuilder()
    while(i >= 0 && j >= 0) {
      val digit1 = num1(i) - '0'
      val digit2 = num2(j) - '0'
      val digit = digit1 + digit2 + carry
      str.append(digit % 10)
      carry = digit / 10
      i -= 1
      j -= 1
    }

    while(i >= 0) {
      val digit = num1(i) - '0' + carry
      str.append(digit % 10)
      carry = digit / 10
      i -= 1
    }

    while(j >= 0) {
      val digit = num2(j) - '0' + carry
      str.append(digit % 10)
      carry = digit / 10
      j -= 1
    }

    if(carry != 0) str.append(carry)
    str.reverse.toString
  }
}

File: src/main/java/com/vc/easy/L415.java
package com.vc.easy;

class L415 {
    public String addStrings(String num1, String num2) {
        int i = num1.length() - 1, j = num2.length() - 1;
        int carry = 0;
        StringBuilder sb = new StringBuilder();
        while(i >= 0 && j >= 0) {
            int chi = num1.charAt(i) - '0';
            int chj = num2.charAt(j) - '0';
            int total = chi + chj + carry;
            int digit = total % 10;
            carry = total / 10;
            sb.insert(0, digit);
            i--;
            j--;
        }

        while(i >= 0) {
            int ch = num1.charAt(i) - '0';
            int total = ch + carry;
            int digit = total % 10;
            carry = total / 10;
            sb.insert(0, digit);
            i--;
        }

        while(j >= 0) {
            int ch = num2.charAt(j) - '0';
            int total = ch + carry;
            int digit = total % 10;
            carry = total / 10;
            sb.insert(0, digit);
            j--;
        }

        if(carry > 0)  sb.insert(0, carry);
        return sb.toString();
    }
}
